server/cmwell-dc/src/main/scala/cmwell/dc/stream/InfotonRetriever.scala

```diff
@@ -53,22 +53,24 @@ object InfotonRetriever extends LazyLogging {
   val initialRetrieveSingleStatus =
     RetrieveStateStatus(Settings.initialSingleRetrieveRetryCount, None)
 
-  case class ExtraData(subject: String, indexTime: Option[Long], lastModifiedBy: Option[String]){
+  case class ExtraData(subject: String, indexTime: Option[Long], lastModifiedBy: Option[String], uuid:Option[String]){
     def shouldMerge (other: ExtraData): Option[ExtraData] = {
       require(other.subject != "", "Subject in ExtraData cannot be empty!")
       require(!(subject!="" && other.subject != subject), "Same path cannot have different subjects!")
 
       other match {
-        case ExtraData(newSubject, None, None) if this.subject == "" => Some(this.copy(subject = newSubject))
-        case ExtraData(_, None, None) => None
-        case ExtraData(newSubject, indTime @ Some(_), None) => Some(this.copy(subject = newSubject, indexTime = indTime))
-        case ExtraData(newSubject, None, lmb @ Some(_)) => Some(this.copy(subject = newSubject, lastModifiedBy = lmb))
+        case ExtraData(newSubject, None, None, None) if this.subject == "" => Some(this.copy(subject = newSubject))
+        case ExtraData(_, None, None, None) => None
+        case ExtraData(newSubject, indTime @ Some(_), None, None) => Some(this.copy(subject = newSubject, indexTime = indTime))
+        case ExtraData(newSubject, None, lmb @ Some(_), None) => Some(this.copy(subject = newSubject, lastModifiedBy = lmb))
+        case ExtraData(newSubject, None, None, uuid @ Some(_)) => Some(this.copy(subject = newSubject, uuid = uuid))
+
       }
     }
 
     def getQuads(indexTimeN: Long, modifierN: Option[String]) = {
       def modifier = modifierN.fold(throw ModifierMissingException(subject))(identity)
-
+      uuid.fold("")(u => s"""<$subject> <cmwell://meta/sys#uuid> "$u"^^<http://www.w3.org/2001/XMLSchema#string> .\n""") ++
```

Comment:
there should ALWAYS by uuid. You should throw an exception if not.

Reply:
done